Workaround for hwloc-induced scylladb crash on ChromeOS

twunderlich-grapl opened the pull request:
<!-- Thank you for your contribution to Grapl! Please take some time
to fill out this short template to help us review and understand your
PR. -->

### Which issue does this PR correspond to?
Closes https://github.com/grapl-security/issue-tracker/issues/912
<!-- Please provide a link to the GitHub Issue this pull request is
meant to address. -->

### What changes does this PR make to Grapl? Why?
Currently there's a chrome bug https://bugs.chromium.org/p/chromium/issues/detail?id=1304418
that causes the crostini vm to send invalid hardware information. This
bug causes scylla to crash on startup with a SIGFPE https://github.com/scylladb/scylla/issues/10439.
Since hwloc has already issues a patch to deal with this, we're
replacing the dynamically linked lib scylladb uses with the patched version
<!-- Please describe the functional impact of these changes and the
rationale behind them. This will help us understand your approach to
solving the problem. -->

### How were these changes tested?

<!-- Please describe how these changes were tested. There should be
sufficient detail that reviewers can replicate your testing
procedure. If the procedure is a manual one that's OK, your
description will help us work with you to get it automated. -->

Commit:
Workaround for hwloc-induced scylladb crash on ChromeOS

Currently there's a chrome bug https://bugs.chromium.org/p/chromium/issues/detail?id=1304418
that causes the crostini vm to send invalid hardware information. This
bug causes scylla to crash on startup with a SIGFPE https://github.com/scylladb/scylla/issues/10439.
Since hwloc has already issues a patch to deal with this, we're
replacing the dynamically linked lib scylladb uses with the patched version

christophermaier commented on scylladb/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+    --tlsv1.2 \
+    --location \
+    --output /usr/local/lib/libhwloc.so.15.5.3 \
+    "https://dl.cloudsmith.io/public/grapl/thomas/raw/versions/15.5.3/libhwloc.so.15.5.3" && \
```
We can (and probably should) put this in one of our "real" repositories (like `releases`), rather than your personal one.

christophermaier commented on scylladb/README.md:
```diff
@@ -0,0 +1,35 @@
+cd hwloc-${HWLOC_VERSION}
+./configure
+make
+make install
```
I was going to suggest creating an actual script for this, so it would be executable. Another possibility would be to fold this into the Dockerfile itself (e.g., in a different target). I'm not sure how long it takes to build, but if it's not too long, it might be good to remove some additional moving parts (including the need to store anything in Cloudsmith, not that that's a horrible thing to do, of course).

colin-grapl replied:
Moving into the Dockerfile as a first stage makes sense to me

twunderlich-grapl replied:
I've moved it into the Dockerfile

christophermaier reviewed (approved):
Couple suggestions, but 👍 overall! Nice solution!

colin-grapl commented on scylladb/Dockerfile:
```diff
@@ -0,0 +1,13 @@
+
+# This downloads a prebuild static lib for hwloc 2.7.1 which includes a chromeos fix, and replaces the dynamically
+# linked version that scylla uses. See the README for more info
+
```
Can we add a comment here to the Scylla issue tracker / our internal tracker here

colin-grapl replied:
Nevermind, I see it below!

colin-grapl commented on scylladb/README.md:
```diff
@@ -0,0 +1,35 @@
+# Summary
+
+This directory includes a dockerfile to create a patched version of scylladb so
+it will run in chromeos. Without it, scylladb
+[crashes with a SIGFPE error from hwloc](https://github.com/scylladb/scylla/issues/10439).
+This is due to a
+[chromeos bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1304418).
+Fortunately, hwloc has released a
+[patch](https://github.com/open-mpi/hwloc/commit/33b555b5a1c8339daeb4215bad57430b57a6b33f)
+in 2.7.1.
```
Hell yeah docs

Commit:
Add scylladb to CODEOWNERS

Commit:
Use Scylladb 4.6.3

Commit:
Use correct name for CODEOWNERS

Commit:
Use a build container for hwloc

Instead of downloading everything from cloudsmith, we build hwloc in
another container. This makes it easy to upgrade as necessary

Commit:
Fix docker lints for scylladb

Commit:
Scylladb dockerfile improvements

Fix the forgotten --directory flag for tar, which caused breakage.
Change tar to use long flags
